Ćwiczenie 1: Series — tworzenie i operacje

In [13]:
import pandas as pd
import numpy as np

# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


In [14]:
#Zadanie 1a: Utwórz Series pensje z dict:
pensje = pd.Series({
    'Anna': 5500,
    'Jan': 7200,
    'Ewa': 4800,
    'Marek': 9100,
    'Kasia': 6300
})

print(pensje)


Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


In [15]:
#Zadanie 1b: Odpowiedz na pytania o pensje:
# 1. kto zarabia najwięcej
print("Najwięcej zarabia:", pensje.idxmax())

# 2. średnia pensja
print("Średnia pensja:", pensje.mean())

# 3. osoby powyżej średniej
print("Powyżej średniej:")
print(pensje[pensje > pensje.mean()])

# 4. ile osób zarabia poniżej 6000
print("Liczba osób < 6000:", (pensje < 6000).sum())


Najwięcej zarabia: Marek
Średnia pensja: 6580.0
Powyżej średniej:
Jan      7200
Marek    9100
dtype: int64
Liczba osób < 6000: 2


In [16]:
#Zadanie 1c: Oblicz pensje po zmianach:
# 1. podwyżka 10%
pensje_10 = pensje * 1.10
print("Po podwyżce 10%:\n", pensje_10)

# 2. premia 500 zł
pensje_premia = pensje + 500
print("\nPo premii 500:\n", pensje_premia)

# 3. nowa pensja (podwyżka + premia)
pensje_nowe = pensje * 1.10 + 500
print("\nNowe pensje:\n", pensje_nowe)

# 4. ranking od najwyższej
ranking = pensje_nowe.sort_values(ascending=False)
print("\nRanking (od najwyższej):\n", ranking)

Po podwyżce 10%:
 Anna      6050.0
Jan       7920.0
Ewa       5280.0
Marek    10010.0
Kasia     6930.0
dtype: float64

Po premii 500:
 Anna     6000
Jan      7700
Ewa      5300
Marek    9600
Kasia    6800
dtype: int64

Nowe pensje:
 Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10510.0
Kasia     7430.0
dtype: float64

Ranking (od najwyższej):
 Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


Ćwiczenie 2: DataFrame

In [17]:
#Krok 1 — DataFrame z dict
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
    'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
    'cena': [3500, 1800, 2500, 350, 2200],
    'stan_magazynowy': [45, 120, 200, 500, 75],
    'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

In [18]:
#Zadanie 2a: Zbadaj DataFrame sklep:
# 1. kształt tabeli
print("Shape:", sklep.shape)

# 2. nazwy kolumn
print("\nKolumny:", sklep.columns)

# 3. typy danych
print("\nTypy danych:\n", sklep.dtypes)

# 4. pierwsze 3 wiersze
print("\nHead:\n", sklep.head(3))


Shape: (5, 4)

Kolumny: Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')

Typy danych:
 produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object

Head:
     produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


In [19]:
#Zadanie 2b: Dodaj nowe kolumny do DataFrame sklep:
# 1. wartość magazynu
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']

# 2. cena z VAT
sklep['cena_z_vat'] = sklep['cena'] * 1.23

# 3. cały DataFrame
print(sklep)


     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


In [20]:
# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

In [21]:
#Zadanie 2c: Podstawowa eksploracja datasetu tips:
# 1. ile wierszy i kolumn
print("Shape:", tips.shape)

# 2. ostatnie 5 wierszy
print("\nTail:\n", tips.tail())

# 3. 3 losowe wiersze
print("\nSample:\n", tips.sample(3))

# 4. typy danych
print("\nDtypes:\n", tips.dtypes)


Shape: (244, 7)

Tail:
      total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

Sample:
      total_bill   tip     sex smoker   day    time  size
49        18.04  3.00    Male     No   Sun  Dinner     2
173       31.85  3.18    Male    Yes   Sun  Dinner     2
125       29.80  4.20  Female     No  Thur   Lunch     6

Dtypes:
 total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


Ćwiczenie 3: Eksploracja datasetu — samodzielna analiza

In [22]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

In [23]:
#Zadanie 3a: Pierwszy rzut oka

# 1. head(10)
print(penguins.head(10))

# 2. shape
print("\nShape:", penguins.shape)

# 3. info()
print("\nInfo:")
penguins.info()

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

In [24]:
#Zadanie 3b: Statystyki opisowe

# 1. describe()
print(penguins.describe())

# 2. describe(all)
print("\nDescribe all:\n", penguins.describe(include='all'))

# 3. mediana masy
print("\nMediana body_mass_g:", penguins['body_mass_g'].median())

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

Describe all:
        species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN      

In [27]:
#Zadanie 3c: Rozkłady wartości
# 1. ile gatunków
print("Species:\n", penguins['species'].value_counts())

# 2. ile na wyspach
print("\nIsland:\n", penguins['island'].value_counts())

# 3. płeć
print("\nSex:\n", penguins['sex'].value_counts())

# 4. % Adelie
print("\n% Adelie:\n", penguins['species'].value_counts(normalize=True))


Species:
 species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

Island:
 island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

Sex:
 sex
MALE      168
FEMALE    165
Name: count, dtype: int64

% Adelie:
 species
Adelie       0.441860
Gentoo       0.360465
Chinstrap    0.197674
Name: proportion, dtype: float64


In [28]:
#Zadanie 3d: Brakujące dane
# 1. brakujące wartości
print("Braki:\n", penguins.isna().sum())

# 2. % braków w sex
braki_sex = penguins['sex'].isna().sum()
procent = braki_sex / len(penguins) * 100

print(f"\n% braków w 'sex': {procent:.2f}%")


Braki:
 species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

% braków w 'sex': 3.20%


Wnioski:
-Dataset zawiera 344 pingwiny z 3 gatunków (Adelie, Chinstrap, Gentoo).
-Najczęstszy gatunek to Adelie (152 osobniki, ok. 44%).
-Pingwiny występują na 3 wyspach, najwięcej na Biscoe (168 osobników).
-W danych występują braki – np. w kolumnie sex brakuje ok. 3% wartości.
-Średnia masa ciała pingwinów wynosi ok. 4200 g, a mediana to ok. 4050 g.
-Dane dotyczące płci nie są idealnie zbalansowane (więcej samców niż samic).

Ćwiczenie 4: Pytania biznesowe + commit

In [29]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')

# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

In [30]:
#Zadanie 4a:
# 1. średni rachunek
print("Średni rachunek:", tips['total_bill'].mean())

# 2. średni napiwek %
print("Średni napiwek %:", tips['tip_pct'].mean())

# 3. który dzień ma najwięcej rachunków
print("\nRachunki wg dnia:\n", tips['day'].value_counts())

# 4. ile rachunków od palaczy
print("\nSmoker:\n", tips['smoker'].value_counts())

# 5. describe dla tip + mediana
print("\nDescribe tip:\n", tips['tip'].describe())
print("Mediana napiwku:", tips['tip'].median())

# 6. % Dinner vs Lunch
print("\nProcent Lunch/Dinner:\n", tips['time'].value_counts(normalize=True) * 100)

# 7. ile unikalnych rozmiarów grup
print("\nUnikalne size:", tips['size'].nunique())


Średni rachunek: 19.78594262295082
Średni napiwek %: 16.08155737704918

Rachunki wg dnia:
 day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64

Smoker:
 smoker
No     151
Yes     93
Name: count, dtype: int64

Describe tip:
 count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64
Mediana napiwku: 2.9

Procent Lunch/Dinner:
 time
Dinner    72.131148
Lunch     27.868852
Name: proportion, dtype: float64

Unikalne size: 6
